tests/fsharp/PlatformHelpers.fs

```diff
@@ -0,0 +1,204 @@
+            input inputWriter
+            do! inputWriter.FlushAsync () |> Async.AwaitIAsyncResult |> Async.Ignore
+            inputWriter.Close ()
+            })
+        |> Option.iter Async.Start
+
+        let exitCode = p |> exitedAsync |> Async.AwaitTask |> Async.RunSynchronously
+
+        match exitCode with
+        | 0 -> Success
+        | err -> ErrorLevel err
+
+
+
+type Result<'S,'F> =
+    | Success of 'S
+    | Failure of 'F
+
+type Attempt<'S,'F> = (unit -> Result<'S,'F>)
+
+open System.Diagnostics
+
+[<DebuggerStepThrough>]
+let internal succeed x = (fun () -> Success x)
+
+[<DebuggerStepThrough>]
+let internal failed err = (fun () -> Failure err)
+
+[<DebuggerStepThrough>]
+let runAttempt (a: Attempt<_,_>) = a ()
+
+[<DebuggerStepThrough>]
+let delay f = (fun () -> f() |> runAttempt)
+
+[<DebuggerStepThrough>]
+let either successTrack failTrack (input : Attempt<_, _>) : Attempt<_, _> =
+    match runAttempt input with
+    | Success s -> successTrack s
+    | Failure f -> failTrack f
+
+[<DebuggerStepThrough>]
+let bind successTrack = either successTrack failed
+
+[<DebuggerStepThrough>] 
+let fail failTrack result = either succeed failTrack result
+
+[<DebuggerStepThrough>] 
+type Attempt =
+    static member Run x = runAttempt x
+
+[<DebuggerStepThrough>] 
+type AttemptBuilder() =
+    member this.Bind(m : Attempt<_, _>, success) = bind success m
+    member this.Bind(m : Result<_, _>, success) = bind success (fun () -> m)
+    member this.Bind(m : Result<_, _> option, success) = 
+        match m with
+        | None -> this.Combine(this.Zero(), success)
+        | Some x -> this.Bind(x, success)
+    member this.Return(x) : Attempt<_, _> = succeed x
+    member this.ReturnFrom(x : Attempt<_, _>) = x
+    member this.Combine(v, f) : Attempt<_, _> = bind f v
+    member this.Yield(x) = Success x
+    member this.YieldFrom(x) = x
+    member this.Delay(f) : Attempt<_, _> = delay f
+    member this.Zero() : Attempt<_, _> = succeed ()
+    member this.While(guard, body: Attempt<_, _>) =
+        if not (guard()) 
+        then this.Zero() 
+        else this.Bind(body, fun () -> 
+            this.While(guard, body))  
+
+    member this.TryWith(body, handler) =
+        try this.ReturnFrom(body())
+        with e -> handler e
+
+    member this.TryFinally(body, compensation) =
+        try this.ReturnFrom(body())
+        finally compensation() 
+
+    member this.Using(disposable:#System.IDisposable, body) =
+        let body' = fun () -> body disposable
+        this.TryFinally(body', fun () -> 
+            match disposable with 
+                | null -> () 
+                | disp -> disp.Dispose())
+
+    member this.For(sequence:seq<'a>, body: 'a -> Attempt<_,_>) =
+        this.Using(sequence.GetEnumerator(),fun enum -> 
+            this.While(enum.MoveNext, 
+                this.Delay(fun () -> body enum.Current)))
+
+let processor = new AttemptBuilder()
```

Comment:
i dont really remember why i changed it to `processor`, `attempt` is better and shorter. Ok for `attempt`